modules/nlu/src/backend/prediction-handler.ts

```diff
@@ -11,11 +11,35 @@ export class PredictionHandler {
     private modelProvider: ModelProvider,
     private engine: sdk.NLU.Engine,
     private anticipatedLanguage: string,
-    private defaultLanguage: string
+    private defaultLanguage: string,
+    private logger: sdk.Logger
   ) {}
 
   async predict(textInput: string, includedContexts: string[]) {
-    const detectedLanguage = await this.engine.detectLanguage(textInput, this.modelsByLang)
+    const modelCacheState = _.mapValues(this.modelsByLang, model => ({ model, loaded: this.engine.hasModel(model) }))
+
+    const missingModels = _(modelCacheState)
+      .toPairs()
+      .filter(([lang, mod]) => !mod.loaded)
+      .fromPairs()
```

Comment:
.fromPairs is not necessary

You can .map('1') and it'll yield the same result

Samething for loaded models

Reply:
:+1:

Reply:
I went for this instead:

```ts
const missingModels = _(modelCacheState)
      .pickBy(mod => !mod.loaded)
      .mapValues(({ model }) => model)
      .value()
```

I really don't like the `.map("someString")` as if the object signature changes, the compiler won't know.